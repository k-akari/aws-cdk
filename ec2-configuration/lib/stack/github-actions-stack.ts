import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class GithubActionsStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new IamRole().createGithubRole(this);
  }
}
