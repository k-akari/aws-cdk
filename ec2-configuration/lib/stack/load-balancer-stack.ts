import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc, CfnSecurityGroup } from 'aws-cdk-lib/aws-ec2';
import { CfnInstance } from 'aws-cdk-lib/aws-ec2';
import { Elb } from '../resource/elb';
import { Subnet } from '../resource/subnet';

export class LoadBalancerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, sg: CfnSecurityGroup, ec2Instance: CfnInstance, props?: StackProps) {
    super(scope, id, props);
   
    new Elb(this, vpc, subnet, sg, ec2Instance);
  }
}
