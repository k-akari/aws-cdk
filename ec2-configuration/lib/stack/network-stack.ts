import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
