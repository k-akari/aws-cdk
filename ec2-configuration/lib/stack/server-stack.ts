import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { Ec2 } from '../resource/ec2';
import { Subnet } from '../resource/subnet';
import { SecurityGroup } from '../resource/security-group';
import { IamRole } from '../resource/iam-role';
import { ElasticIp } from '../resource/elastic-ip';
import { Elb } from '../resource/elb';

export class ServerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, subnet: Subnet, props?: StackProps) {
    super(scope, id, props);

    const sg = new SecurityGroup(this, vpc);

    const instanceProfile = new IamRole().createEc2InstanceProfile(this);
    const ec2Instance = new Ec2().createEc2Instance(this, subnet.public1a, instanceProfile, sg.ec2);
    new ElasticIp().associateElasticIp(this, ec2Instance, 'ec2-1a');

    new Elb(this, vpc, [subnet.public1a, subnet.public1c], sg.alb, ec2Instance);
  }
}
