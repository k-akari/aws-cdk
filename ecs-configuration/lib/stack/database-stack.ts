import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Subnet } from '../resource/subnet';
import { Rds } from '../resource/rds';
import { SecurityGroup } from '../resource/security-group';
import { SecretsManager } from '../resource/secrets-manager';

export class DatabaseStack extends Stack {
  constructor(scope: Construct, id: string, subnet: Subnet, sg: SecurityGroup, ssm: SecretsManager, props?: StackProps) {
    super(scope, id, props);
   
    new Rds(this, subnet, sg, ssm);
  }
}
