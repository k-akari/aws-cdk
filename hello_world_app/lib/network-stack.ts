import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internetGateway';
import { ElasticIp } from './resource/elasticIp';
import { NatGateway } from './resource/natGateway';
import { RouteTable } from './resource/routeTable';
import { NetworkAcl } from './resource/networkAcl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    // VPC
    const vpc = new Vpc();
    vpc.createResources(this);

    // Subnet
    const subnet = new Subnet(vpc.vpc);
    subnet.createResources(this);

    // Internet Gateway
    const internetGateway = new InternetGateway(vpc.vpc);
    internetGateway.createResources(this);

    // Elastic IP
    const elasticIp = new ElasticIp();
    elasticIp.createResources(this);

    // NAT Gateway
    const natGateway = new NatGateway(
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      elasticIp.ngw1a,
      elasticIp.ngw1c
    );
    natGateway.createResources(this);

    // Route Table
    const routeTable = new RouteTable(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c,
      internetGateway.igw,
      natGateway.ngw1a,
      natGateway.ngw1c
    );
    routeTable.createResources(this);

    // Network ACL
    const networkAcl = new NetworkAcl(
      vpc.vpc,
      subnet.subnetPublic1a,
      subnet.subnetPublic1c,
      subnet.subnetPrivate1a,
      subnet.subnetPrivate1c
    );
    networkAcl.createResources(this);
  }
}
