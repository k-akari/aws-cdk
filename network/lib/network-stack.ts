import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from "aws-cdk-lib/aws-ec2";

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    const vpc = new Vpc(this, "sample-vpc", {});
  }
}
