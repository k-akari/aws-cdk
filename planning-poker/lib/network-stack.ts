import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internet-gateway';
import { ElasticIp } from './resource/elastic-ip';
import { NatGateway } from './resource/nat-gateway';
import { RouteTable } from './resource/route-table';
import { NetworkAcl } from './resource/network-acl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new Vpc(this);
    const subnet = new Subnet(this, vpc);
    const internetGateway = new InternetGateway(this, vpc);
    const elasticIp = new ElasticIp(this);
    const natGateway = new NatGateway(this, subnet, elasticIp);
    new RouteTable(this, vpc, subnet, internetGateway, natGateway);
    new NetworkAcl(this, vpc, subnet);
  }
}
