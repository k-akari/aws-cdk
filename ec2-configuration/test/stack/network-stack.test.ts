import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as Network from '../../lib/stack/network-stack';

test('NetworkStackSnapshot', () => {
  const serviceName = 'service';
  const envType = 'test';
  const app = new App({
    context: {
      'serviceName': serviceName,
      'envType': envType
    }
  });
  const stack = new Network.NetworkStack(app, 'NetworkStack');
  const template = Template.fromStack(stack);

  expect(template).toMatchSnapshot();
});
