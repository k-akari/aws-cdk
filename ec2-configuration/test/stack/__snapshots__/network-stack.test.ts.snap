// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NetworkStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "AssociationPrivate1a": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePrivate1a",
        },
        "SubnetId": Object {
          "Ref": "SubnetPrivate1a",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AssociationPublic1a": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePublic",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublic1a",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "AssociationPublic1c": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "RouteTablePublic",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublic1c",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "ElasticIpNgw1a": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-eip-ngw-1a",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "InternetGateway": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-igw",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "NatGateway1a": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "ElasticIpNgw1a",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "SubnetPublic1a",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-ngw-1a",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "NetworkAclAssociationPrivate1a": Object {
      "Properties": Object {
        "NetworkAclId": Object {
          "Ref": "NetworkAclPrivate",
        },
        "SubnetId": Object {
          "Ref": "SubnetPrivate1a",
        },
      },
      "Type": "AWS::EC2::SubnetNetworkAclAssociation",
    },
    "NetworkAclAssociationPublic1a": Object {
      "Properties": Object {
        "NetworkAclId": Object {
          "Ref": "NetworkAclPublic",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublic1a",
        },
      },
      "Type": "AWS::EC2::SubnetNetworkAclAssociation",
    },
    "NetworkAclAssociationPublic1c": Object {
      "Properties": Object {
        "NetworkAclId": Object {
          "Ref": "NetworkAclPublic",
        },
        "SubnetId": Object {
          "Ref": "SubnetPublic1c",
        },
      },
      "Type": "AWS::EC2::SubnetNetworkAclAssociation",
    },
    "NetworkAclEntryInboundPrivate": Object {
      "Properties": Object {
        "CidrBlock": "0.0.0.0/0",
        "Egress": false,
        "NetworkAclId": Object {
          "Ref": "NetworkAclPrivate",
        },
        "Protocol": -1,
        "RuleAction": "allow",
        "RuleNumber": 100,
      },
      "Type": "AWS::EC2::NetworkAclEntry",
    },
    "NetworkAclEntryInboundPublic": Object {
      "Properties": Object {
        "CidrBlock": "0.0.0.0/0",
        "Egress": false,
        "NetworkAclId": Object {
          "Ref": "NetworkAclPublic",
        },
        "Protocol": -1,
        "RuleAction": "allow",
        "RuleNumber": 100,
      },
      "Type": "AWS::EC2::NetworkAclEntry",
    },
    "NetworkAclEntryOutboundPrivate": Object {
      "Properties": Object {
        "CidrBlock": "0.0.0.0/0",
        "Egress": true,
        "NetworkAclId": Object {
          "Ref": "NetworkAclPrivate",
        },
        "Protocol": -1,
        "RuleAction": "allow",
        "RuleNumber": 100,
      },
      "Type": "AWS::EC2::NetworkAclEntry",
    },
    "NetworkAclEntryOutboundPublic": Object {
      "Properties": Object {
        "CidrBlock": "0.0.0.0/0",
        "Egress": true,
        "NetworkAclId": Object {
          "Ref": "NetworkAclPublic",
        },
        "Protocol": -1,
        "RuleAction": "allow",
        "RuleNumber": 100,
      },
      "Type": "AWS::EC2::NetworkAclEntry",
    },
    "NetworkAclPrivate": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-nacl-private",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::NetworkAcl",
    },
    "NetworkAclPublic": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-nacl-public",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::NetworkAcl",
    },
    "RoutePrivate1a": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "NatGateway1a",
        },
        "RouteTableId": Object {
          "Ref": "RouteTablePrivate1a",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "RoutePublic": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "InternetGateway",
        },
        "RouteTableId": Object {
          "Ref": "RouteTablePublic",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "RouteTablePrivate1a": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-rtb-private-1a",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "RouteTablePublic": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-rtb-public",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "SubnetPrivate1a": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.0.100.0/24",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-subnet-private-1a",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "SubnetPublic1a": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.0.10.0/24",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-subnet-public-1a",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "SubnetPublic1c": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "10.0.11.0/24",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-subnet-public-1c",
          },
        ],
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "Vpc8378EB38": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "service-test-vpc",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "VpcGatewayAttachment": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "InternetGateway",
        },
        "VpcId": Object {
          "Ref": "Vpc8378EB38",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
